syntax = "proto2";
package ear.plugin.proto;
import "common_types.proto";

message GainInteractive {
  optional bool enabled = 1 [default = false];
  optional float min = 2 [default = 0.5];
  optional float max = 3 [default = 2.0];
}

message OnOffInteractive {
  optional bool enabled = 1 [default = false];
}

message PositionInteractive {
  optional bool enabled = 1 [default = false];
  optional float min_az = 2 [default = -30.0];
  optional float max_az = 3 [default = 30.0];
  optional float min_el = 4 [default = -30.0];
  optional float max_el = 5 [default = 30.0];
  optional float min_r = 6 [default = 0.0];
  optional float max_r = 7 [default = 0.0];
}

message Object {
  optional string connection_id = 1  [default = "00000000-0000-0000-0000-000000000000"];
  optional bool show_settings = 2 [default = false];
  optional OnOffInteractive interactive_on_off = 3;
  optional GainInteractive interactive_gain = 4;
  optional PositionInteractive interactive_position = 5;
}
message Toggle {
  optional string name = 1 [default = "Toggle"];
  optional string language = 2 [default = ""];
  optional int32 default_element_index = 3 [default = -1];
  optional int32 selected_element_index = 4 [default = -1];
  repeated ProgrammeElement element = 5;
}
message Group {
  optional string name = 1 [default = "Group"];
  optional string language = 2 [default = ""];
  optional ContentKind content_kind = 3 [default = UNKNOWN];
  repeated ProgrammeElement element = 4;
}
message ProgrammeElement {
  oneof item {
    Object object = 1;
    Toggle toggle = 2;
    Group group = 3;
  }
}
message Programme {
  optional string name = 1 [default = "Default"];
  optional string language = 2 [default = ""];
  repeated ProgrammeElement element = 3;
  optional string programme_internal_id = 4  [default = "00000000-0000-0000-0000-000000000000"];
}
message ProgrammeStore {
  repeated Programme programme = 1;
  optional string selected_programme_internal_id = 2  [default = "00000000-0000-0000-0000-000000000000"];
  optional bool auto_mode = 3 [default = true];
}
